### PR TITLE
feat: add 404 page

### LINKED ISSUE #90: BUG: Some pages have no content
Some pages show nothing:

- `/crate`
- `/repo`
- `/repo/github`

There could be a link back to the home page or a 404 error.

### PR DESCRIPTION
**Pull request by Feliix42**
This small PR introduces a very simple `404 error` page to indicate to the user that something went wrong rather than producing just a blank page, as before. This should close #90.

### CODE REVIEW

**Comment by robjtede** [comment]
afk... screenshot?

**Comment by Feliix42** [comment]
This is what the 404 page looks like:
404 - Deps rs

**Comment by robjtede** [comment]
can set this page to have some light caching maybe ?

**Review by robjtede — APPROVED** [review]

File: src/server/views/html/error.rs [hunk 1]
```diff
@@ -20,3 +22,39 @@ pub fn render(title: &str, descr: &str) -> Response<Body> {
...
+                }
+                (super::render_footer(None))
+            }
+        }
+    };
+
+    Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .header(CONTENT_TYPE, "text/html; charset=utf-8")
+        .body(Body::from(rendered.0))
```
Comment (Enet4) [inline src/server/views/html/error.rs:58]: Unless anyone's opposed with this, it should be safe to tell the client to cache this page. To the best of my knowledge, rendered views are not cached by the server.
```suggestion
        .header(CACHE_CONTROL, "public, max-age=365000000, immutable")
        .body(Body::from(rendered.0))
```

Comment (paolobarbolini) [reply src/server/views/html/error.rs:58]: This seems far too long. I would limit it to only a few minutes.

Comment (Feliix42) [reply src/server/views/html/error.rs:62]: I set the caching time to 5 minutes which seems like a sane default.

**Review by Enet4 — APPROVED** [review]

**Pull request merged**